### PR TITLE
Improve accessibility support a bit

### PR DESCRIPTION
**Pull request by hunger**
* Follow up on last PR
* Make Button and tab widget accessible in the fluent style

### CODE REVIEW

File: internal/compiler/widgets/fluent/std-widgets.slint [hunk 1]
```diff
@@ -338,6 +342,10 @@ export TabImpl := Rectangle {
     horizontal-stretch: 0;
     vertical-stretch: 0;
 
+    accessible-role: tab;
+    accessible-has-focus <=> has-focus;
+    accessible-label <=> title;
```
Comment (tronical) [inline internal/compiler/widgets/fluent/std-widgets.slint:347]: I really like this API and approach.

Implementation wise, will these two-way bindings end up as aliases, where the `accessible_string_property` implementation will end up querying the title property? (that's how it looks like to me, but I'm curious if I read that right)

File: internal/common/enums.rs [hunk 2]
```diff
@@ -180,9 +180,10 @@ macro_rules! for_each_enums {
             /// The role for the `accessible-role` property
             enum AccessibleRole {
                 none,
-                text,
                 button,
                 checkbox,
+                tab,
+                text,
```
Comment (ogoffart) [inline internal/common/enums.rs:188]: This is documented in builtin_elements.md

(yeah, ideally we would have the documentation inline and we'd extract the enum documentation)

File: internal/compiler/widgets/fluent/std-widgets.slint (hunk 1 above)
Comment (ogoffart) [reply internal/compiler/widgets/fluent/std-widgets.slint:347]: The  accessible-* property should be optimized away, and the `accessible_string_property` should just end up querying the `title` property.  (modulo bugs)

**Review by ogoffart — COMMENTED** [review]
>  internal/core/accessibility.rs#L24 :  cannot find macro `vec` in this scope

so it is still missing a use of `vec` in internal/core/accessibility.rs

**Comment by hunger** [comment]
Looks like I lost the vec change when rebasing to the updated branch or something:-/ I'll add it back in.

**Comment by hunger** [comment]
Latest push adds entries for the new accessible-roles to builtin_elements.md.

File: internal/backends/qt/qt_accessible.rs [hunk 3]
```diff
@@ -7,9 +7,12 @@ use cpp::*;
 
 use i_slint_core::item_tree::{ItemRc, ItemWeak};
 use i_slint_core::SharedVector;
+
+use pin_project::pin_project;
 use qttypes::QString;
 
 use alloc::boxed::Box;
+use alloc::vec::Vec;
```
Comment (ogoffart) [inline internal/backends/qt/qt_accessible.rs:15]:
```suggestion
use alloc::vec;
```

Comment (ogoffart) [inline internal/backends/qt/qt_accessible.rs:15]: It's in the wrong file :-)  (and its `alloc::vec` for the macro)

**Comment by hunger** [comment]
I should set up a proper MCU build locally:-/

File: internal/compiler/widgets/fluent/std-widgets.slint (hunk 1 above)
Comment (hunger) [reply internal/compiler/widgets/fluent/std-widgets.slint:347]: I think that is what's happening, yes.

Technically these are all sting properties when querying them from the rust code later, even though they are treated as bool (or popentially other types) in slint.

**Comment by ogoffart** [comment]
> I should set up a proper MCU build locally:-/

That's what CI is for.

**Review by ogoffart — APPROVED** [review]

**Pull request merged**